### PR TITLE
kubernetes/operator: bump python edot version in instrumenting-applications

### PR DESCRIPTION
**Pull request by xrmx**

### CODE REVIEW

**Comment by eedugon** [comment]
Thanks a lot @xrmx
The `values.yaml` of the repository still points to `0.3.0`, and that's what the users will be using. Should we also update our `values.yaml` to 0.4.0 for python?

@gizas , @rogercoll : I assume that at some point we should apply `values.yaml` changes in `main` without backporting to `8.16`, right? To avoid breaking something unexpectedly.

Also I think we should make clear in the docs (instrumenting-applications), that the provided details about the `Instrumentation` object is just an example reference and not really the latest or expected version, as that should be in the `values.yaml`.

What do you think? Anyway more than happy to change this value in the doc.

**Comment by jackshirazi** [comment]
- Doc and values.yml should change in sync. 
- Afaict, 8.16 is now stable so should only have bug fixes rather than feature changes (bumping the version is a feature change)
- Yes definitely add a note about upgrading versions since we have a 8.16 branch which will be pointing at out of date docker images, the longer we go the more out of date

**Comment by eedugon** [comment]
thanks @jackshirazi ! the upgrade instructions are part of https://github.com/elastic/opentelemetry/pull/50, feel free to review and comment out.

**Comment by xrmx** [comment]
0.4.0 has features that requires operator 0.113.0 so if we are suggesting an older operator for 8.16 no big deal :)

**Review by jackshirazi — APPROVED** [review]

**Comment by eedugon** [comment]
@xrmx , I've got one question about this change:

> 0.4.0 has portability features that requires operator 0.113.0 

If using 0.4.0 version of EDOT Python requires 0.113.0 version of the operator I think we shouldn't merge this until the operator is updated in our documented installation, because at the moment we are deploying kube-stack 0.3.3 version which comes with operator 0.107.0 if i'm not mistaken.

But please feel free to correct me if I'm wrong, @jackshirazi or @rogercoll .

**Comment by xrmx** [comment]
> @xrmx , I've got one question about this change:
> 
> > 0.4.0 has portability features that requires operator 0.113.0
> 
> If using 0.4.0 version of EDOT Python requires 0.113.0 version of the operator I think we shouldn't merge this until the operator is updated in our documented installation, because at the moment we are deploying kube-stack 0.3.3 version which comes with operator 0.107.0 if i'm not mistaken.

It does not require 0.113.0, it adds support for musl based containers that to be used requires an annotation that has been implemented in operator 0.113.0

**Comment by eedugon** [comment]
Thanks @xrmx , then I think we should clearly merge this.

File: docs/kubernetes/operator/instrumenting-applications.md [hunk 1]
```diff
@@ -80,7 +80,7 @@ The `Instrumentation` object stores important parameters:
   nodejs:
     image: docker.elastic.co/observability/elastic-otel-node:0.4.1
   python:
-    image: docker.elastic.co/observability/elastic-otel-python:0.3.0
+    image: docker.elastic.co/observability/elastic-otel-python:0.4.0
```
Comment (xrmx) [inline docs/kubernetes/operator/instrumenting-applications.md:83]:
```suggestion
    image: docker.elastic.co/observability/elastic-otel-python:0.4.1
```

File: resources/kubernetes/operator/helm/values.yaml [hunk 2]
```diff
@@ -948,6 +948,6 @@ instrumentation:
   dotnet:
     image: docker.elastic.co/observability/elastic-otel-dotnet:edge
   python:
-    image: docker.elastic.co/observability/elastic-otel-python:0.3.0
+    image: docker.elastic.co/observability/elastic-otel-python:0.4.0
```
Comment (xrmx) [inline resources/kubernetes/operator/helm/values.yaml:951]:
```suggestion
    image: docker.elastic.co/observability/elastic-otel-python:0.4.1
```

**Comment by xrmx** [comment]
Bumped to 0.4.1 in the meantime :)

**Pull request merged**